src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -152,6 +152,30 @@ struct DifferentiationMatrixGenerator {
   }
 };
 
+template <Basis BasisType, Quadrature QuadratureType>
+struct WeakFluxDifferentiationMatrixGenerator {
+  Matrix operator()(const size_t num_points) const noexcept {
+    if (BasisType != Basis::Legendre) {
+      ERROR(
+          "Weak differentiation matrix only implemented for Legendre "
+          "polynomials.");
+    }
```

Comment:
For my understanding: why is this a run-time error and not a static_assert? Is it because of how we automatically instantiate the different basis/quadrature combinations, and so we need to be able to instantiate this even for a basis we don't want?

Reply:
That's right. I've added a comment in the code :)